homeassistant/components/nut/__init__.py

```diff
@@ -61,7 +61,9 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     async def async_update_data():
         """Fetch data from NUT."""
         async with async_timeout.timeout(10):
-            return await hass.async_add_executor_job(data.update)
+            await hass.async_add_executor_job(data.update)
+            if not data.status:
+                raise UpdateFailed("Error fetching UPS state")
 
```

Comment:
We're not returning anything from `async_update_data` now. How will the coordinator data get updated?

Reply:
We never look at the coordinator data and are only relying on it to do the updates as everything is in:

```
class PyNUTData:
    """Stores the data retrieved from NUT.
```

All the sensors look at:
`self._data.status`

Reply:
Ok. :+1: